### PR TITLE
Updated graph function so that it works with multiple data inputs

### LINKED ISSUE #10: Create Basic Graph Component for WebApp
We need some sort of graph that can read from a websocket and produce a nice looking graph. This should look like a function that you run that takes in:
* A div id
* A websocket
* Some flags that control what the websocket data represents. i.e. the first 2 are data, the last is the label

This function should then start up whatever is needed to read from the websocket and update the graph. It should also handle the websocket closing unexpectedly. More discussion is needed for what this graph looks for.

Would be used in #8 and #9

### PR DESCRIPTION
**Pull request by chenfeiyu132**
#10

### CODE REVIEW

File: server/res/graph.html [hunk 1]
```diff
@@ -13,62 +13,127 @@
...
         <script>
-            var last_data = 0
-            var last_label = 0
+
+            var datalist = {};
+            var currentGraph = 0;
 
             console.log("LOG LOG LOG")
-            var feedSocket2 = new WebSocket(getWebSocketURI("/ws/ingread/10"));
+            var feedSocket2 = new WebSocket(getWebSocketURI("/ws/data/random"));
```
Comment (cwlowder) [inline server/res/graph.html]: Why is this the second feed socket?

File: server/res/graph.html [hunk 2]
```diff
@@ -13,62 +13,127 @@
       <div class="wrapper">
         <div id="tester"></div>
         <script>
```
Comment (cwlowder) [inline server/res/graph.html:16]: I'd prefer if we put this code into a file called graph.js. That way other htmls can use it. You will need to add a static definition for this in app.py

File: server/res/graph.html [hunk 3]
```diff
@@ -13,62 +13,127 @@
...
+                for(key in datalist) {
+                  newdata = getData(key);
+                  console.log(newdata)
+                  if(newdata != '0' && key !== "flag") {
+                    Plotly.extendTraces(divId, {y: [[newdata]]}, [index]);
+                    //Plotly.update(divId, datalist[key], layout, index)
+                  }
+                  index++;
+                  counter++;
+                  // if(counter > 50) {
```
Comment (cwlowder) [inline server/res/graph.html]: Delete commented out code

File: server/res/graph.html [hunk 4]
```diff
@@ -13,62 +13,127 @@
...
-                last_label = parseInt(msg.data.split(",")[1]);
+              // Do we assume that the configs always mach the number of data in the message?
+              websocket.onmessage = function(msg) {
+                console.log(msg)
+                msg_data = msg.data.split(",");
+                var index = 0;
+                for(i in configs) {
+                  let config = configs[i]
+                  if(index < msg_data.length) {
+                    datalist[config].push(msg_data[index]);
```
Comment (cwlowder) [inline server/res/graph.html]: You should do the remove of old data here. So when the size of the data > 500, start removing elements?

**Review by cwlowder — CHANGES_REQUESTED** [review]

File: server/res/graph.html [hunk 5]
```diff
@@ -13,62 +13,127 @@
       <div class="wrapper">
         <div id="tester"></div>
         <script>
-            var last_data = 0
-            var last_label = 0
+
+            var datalist = {};
+            var currentGraph = 0;
 
             console.log("LOG LOG LOG")
```
Comment (cwlowder) [inline server/res/graph.html]: We should remove this console.log

File: server/res/graph.html [hunk 6]
```diff
@@ -13,62 +13,127 @@
...
+              Plotly.plot(divId, dataset, layout, {displayModeBar: false});
+
+              setInterval(function(){
+                let index = 0;
+                for(key in datalist) {
+                  newdata = getData(key);
+                  console.log(newdata)
+                  if(newdata != '0' && key !== "flag") {
+                    Plotly.extendTraces(divId, {y: [[newdata]]}, [index]);
+                    //Plotly.update(divId, datalist[key], layout, index)
```
Comment (cwlowder) [inline server/res/graph.html]: Remove this comment

File: server/res/graph.html [hunk 7]
```diff
@@ -13,62 +13,127 @@
...
+                  type: 'scatter'
+                })
+              }
+              Plotly.plot(divId, dataset, layout, {displayModeBar: false});
+
+              setInterval(function(){
+                let index = 0;
+                for(key in datalist) {
+                  newdata = getData(key);
+                  console.log(newdata)
```
Comment (cwlowder) [inline server/res/graph.html]: Remove this

File: server/res/graph.html [hunk 8]
```diff
@@ -13,62 +13,127 @@
...
+                    //}
+                  }
+              },60);
 
-            feedSocket2.onmessage = function(msg) {
-                last_data = parseInt(msg.data.split(",")[0]);
-                last_label = parseInt(msg.data.split(",")[1]);
+              // Do we assume that the configs always mach the number of data in the message?
+              websocket.onmessage = function(msg) {
+                console.log(msg)
```
Comment (cwlowder) [inline server/res/graph.html]: Maybe remove this too?

**Review by cwlowder — CHANGES_REQUESTED** [review]

File: server/res/graph.html (hunk 1 above)
Comment (chenfeiyu132) [reply server/res/graph.html]: Because you named it this way

File: server/res/graph.html (hunk 8 above)
Comment (chenfeiyu132) [reply server/res/graph.html]: Nah

Comment (chenfeiyu132) [reply server/res/graph.html]: It's useful for debugging

**Review by cwlowder — CHANGES_REQUESTED** [review]
Change max number to 100

**Review by cwlowder — APPROVED** [review]
Looks great!

**Pull request merged**